src/histogram.c

```diff
@@ -192,8 +192,8 @@ ts_hist_deserializefunc(PG_FUNCTION_ARGS)
 	serialized = PG_GETARG_BYTEA_P(0);
 
 	buf.data = VARDATA(serialized);
-	buf.len = VARSIZE(serialized);
-	buf.maxlen = VARSIZE(serialized);
+	buf.len = VARSIZE(serialized) - VARHDRSZ;
```

Comment:
I guess it would be good to have a test that could trigger the bad behavior without this fix.

Reply:
Since this is a stringinfo data type, we should use the API `appendBinaryStringInfo` to set these fields instead of directly modifying the struct.

Reply:
I missed the maxlen, but still think it is good to use stringinfo functions. I don't think the extra copy is a problem.